src/System.Private.CoreLib/shared/System/Exception.cs

```diff
@@ -159,5 +160,15 @@ public int HResult
         public new Type GetType() => base.GetType();
 
         partial void RestoreRemoteStackTrace(SerializationInfo info, StreamingContext context);
+
+        /// <summary>Stores the current stack trace into this exception instance.</summary>
+        /// <param name="throwIfHasExistingStack">
+        /// true to throw if the exception already has a stack; false to nop if the exception already has a stack.
+        /// </param>
+        [StackTraceHidden]
+        internal void SetCurrentStackTrace(bool throwIfHasExistingStack) => SetCurrentStackTraceCore(throwIfHasExistingStack);
```

Comment:
I do not think we need this wrapper. `internal void SetCurrentStackTrace(...)` can be directly in Exception.CoreCLR.cs.

Reply:
partial methods have to be private.  What would you recommend instead?

Reply:
Delete SetCurrentStackTrace from here. Rename SetCurrentStackTraceCore to SetCurrentStackTrace in Exception.cs.

Reply:
Sorry, I'm not following.  SetCurrentStackTraceCore is partial; it has to be private.  And as private, it can't be called from ExceptionDispatchInfo.  If I make it not partial, then I can't specialize it for coreclr and would instead need another method in Exception.CoreCLR.cs it can call, along with an equivalent method in other partials in other runtimes.  Am I misunderstanding?

Reply:
>  I'll similarly need to add that same method to other partial files in other runtimes

Yes, I think other runtimes should implement this method as well (it is ok to have a empty impl with TODO).